flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/stream/sql/SortTest.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.table.api.scala.stream.sql
+
+import org.apache.flink.api.scala._
+import org.apache.flink.table.api.TableException
+import org.apache.flink.table.api.scala._
+import org.apache.flink.table.utils.TableTestUtil._
+import org.apache.flink.table.utils.{StreamTableTestUtil, TableTestBase}
+import org.junit.Test
+
+class SortTest extends TableTestBase {
+  private val streamUtil: StreamTableTestUtil = streamTestUtil()
+  streamUtil.addTable[(Int, String, Long)]("MyTable", 'a, 'b, 'c,
+      'proctime.proctime, 'rowtime.rowtime)
+  
+  @Test
+  def testSortProcessingTime() = {
+
+    val sqlQuery = "SELECT a FROM MyTable ORDER BY proctime, c"
+      
+    val expected =
+      unaryNode(
+        "DataStreamCalc",
+        unaryNode("DataStreamSort",
+          streamTableNode(0),
+          term("orderBy", "proctime ASC", "c ASC"),
+          term("offset", "null"), 
+          term("fetch","unlimited")),
+        term("select", "a", "TIME_MATERIALIZATION(proctime) AS proctime", "c"))
```

Comment:
yes and no :)
Perhaps you remember that in the previous PR I pointed this issue out. Calcite as it creates the execution plan now will have as the output type of the sort a tuple for each of the 3 fields (a, proctime, and c)...because one comes from the output and 2 are used internally. The sort does not do any projection so all the 3 fields needed in the input are carried in the output - hence the type of the output. I have asked you at that point if you believe we should drop the fields by ourselves - which you said no and I agree with that. This test is only the reflection of that case - basically the whole plan of calcite should be modified....